tensorflow/python/framework/tensor_shape.py

```diff
@@ -428,6 +429,8 @@ def __init__(self, dims):
       TypeError: If dims cannot be converted to a list of dimensions.
     """
     # TODO(irving): Eliminate the single integer special case.
+    from tensorflow.python.util import compat
```

Comment:
imports should be at the top of the file in tensorflow

Reply:
`tensor_shape` and `compat` are depended by each other if imports are at the top. I don't know how to deal with the circular imports.